models/classes/class.QtiTestService.php

```diff
@@ -866,24 +869,62 @@ protected function importTest(
         return $report;
     }
 
+    /**
+     * @throws common_Exception
+     */
     private function deleteTestsFromClassByLabel(
         string $testLabel,
         string $itemsClassLabel,
         core_kernel_classes_Resource $testClass,
         core_kernel_classes_Class $itemClass
     ): void {
-        $testService = $this->getTestService();
-        $itemTreeService = $this->getItemTreeService();
+        $itemClassesToDelete = $this->getSubclassesByLabel($itemClass, $itemsClassLabel);
+
+        /** @var string[] $refs */
+        $refs = $this->collectResourceReferences($itemClassesToDelete);
```

Comment:
You do not need to type hint here with annotations, instead, add annotations to `collectResourceReferences`, otherwise every place calling this method will have to be typehinted.

Reply:
Thanks!